modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase

```diff
@@ -1400,4 +1400,318 @@ definition {
+					uomKey = "uom${index}",
+					uomName = "UOM${index}");
+			}
+
+			CommerceJSONSKUsAPI._addCommerceUOM(
+				basePrice = 3,
+				incrementalOrderQuantity = "0.6",
+				productSKU = "MIN93016B",
+				uomKey = "uom3",
+				uomName = "UOM3");
+		}
+
+		task ("And Brake Fluid's Multiple Order Quantity is set to 0.1") {
+			CommerceJSONProductsAPI._patchCommerceProductConfiguration(
+				multipleOrderQuantity = "0.1",
+				productName = "Brake Fluid");
+		}
+
+		task ("And adds to cart MIN93016A/uom1 with quantity 2, MIN93016A/uom2 with quantity 3, MIN93016B/uom3 with quantity 1.2, MIN93016C with quantity 1") {
+			CommerceJSONAccountsAPI._addCommerceAccountAddress(
+				accountEntryName = "Commerce Account",
+				city = "Test City",
+				countryISOCode = "US",
+				name = "Test Address",
+				street1 = "Test Street",
+				zip = "Test Zip Code");
+
+			CommerceJSONDeliveryCart._addCommerceCart(
+				accountEntryName = "Commerce Account",
+				billingAddressName = "Test Address",
+				cartItems = "MIN93016A:2:uom1,MIN93016A:3:uom2,MIN93016C:1,MIN93016B:1.2:uom3",
+				shippingAddressName = "Test Address",
+				shippingOption = "Standard Delivery");
+		}
+
+		task ("And navigates to the order from the admin order page") {
+			CommerceOrders.openOrdersAdmin();
+
+			CommerceNavigator.gotoOrderEntryViaAdmin(status = "Open");
+		}
+
+		task ("Then the UOMs keys and quantities are all correct") {
+			for (var rowList : list "9:MIN93016A:2.0,10:MIN93016A:uom1,9:MIN93016A:3.0,10:MIN93016A:uom2,9:MIN93016B:1.2,10:MIN93016B:uom3,9:MIN93016C:1.00,10:MIN93016C:") {
+				var columnIndex = StringUtil.extractFirst(${rowList}, ":");
+
+				var rowList = StringUtil.replaceFirst(${rowList}, "${columnIndex}:", "");
+
+				var sku = StringUtil.extractFirst(${rowList}, ":");
+				var value = StringUtil.extractLast(${rowList}, ":");
+
+				AssertElementPresent(
+					key_entryTitle = ${sku},
+					key_index = ${columnIndex},
+					key_value = ${value},
+					locator1 = "CommerceEntry#TABLE_ENTRY_FIELD_VALUE");
+			}
+		}
+
+		task ("When the admin updates the precision of uom3 to be 3") {
+			CommerceJSONSKUsAPI._addCommerceUOM(
+				basePrice = 3,
+				incrementalOrderQuantity = "0.6",
+				precision = 3,
+				productSKU = "MIN93016B",
+				uomKey = "uom3",
+				uomName = "UOM3");
+		}
+
+		task ("Then the order item quantity is updated to be 1.200") {
+			CommerceOrders.openOrdersAdmin();
+
+			CommerceNavigator.gotoOrderEntryViaAdmin(status = "Open");
+
+			for (var rowList : list "9:MIN93016B:1.200,10:MIN93016B:uom3") {
+				var columnIndex = StringUtil.extractFirst(${rowList}, ":");
+
+				var rowList = StringUtil.replaceFirst(${rowList}, "${columnIndex}:", "");
+
+				var sku = StringUtil.extractFirst(${rowList}, ":");
+				var value = StringUtil.extractLast(${rowList}, ":");
+
+				AssertElementPresent(
+					key_entryTitle = ${sku},
+					key_index = ${columnIndex},
+					key_value = ${value},
+					locator1 = "CommerceEntry#TABLE_ENTRY_FIELD_VALUE");
+			}
```

Comment:
Here I think you can do this without the for cycle, just 2 assertions with fixed columnIndex